### ISSUE TITLE
Dot Matrix Fonts and Tesseract's Connected Component Analysis

### ISSUE THREAD

**Issue by ameera3** [tesseract-ocr/tesseract#2342]
Tesseract Version:
tesseract 4.0.0-beta.1
leptonica-1.75.3
libgif 5.1.4 : libjpeg 8d (libjpeg-turbo 1.5.2) : libpng 1.6.34 : libtiff 4.0.9 : zlib 1.2.11 : libwebp 0.6.1 : libopenjp2 2.3.0

Found AVX512BW
Found AVX512F
Found AVX2
Found AVX
Found SSE

Platform: uname -a output: Linux CUDA1 4.15.0-29-generic #31~16.04.1-Ubuntu SMP Wed Jul 18 08:54:04 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

cat /etc/lsb-release
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=18.04
DISTRIB_CODENAME=bionic
DISTRIB_DESCRIPTION="Ubuntu 18.04 LTS"

### Current Behavior:  I am trying to fine-tune Tesseract for dot-matrix fonts such as that in the attached picture.
![ex_20](https://user-images.githubusercontent.com/36647759/54852785-1c559480-4cab-11e9-9e78-aa1b14802c37.jpg)

When the dots are closely spaced together and touch, Tesseract can more or less handle the dot-matrix font with some fine-tuning and image processing.  However, when the dots do not touch, as in the picture, Tesseract struggles.

I read in [An Overview of the Tesseract OCR Engine](https://static.googleusercontent.com/media/research.google.com/en//pubs/archive/33418.pdf) that the first step in Tesseract's processing pipeline is a connected component analysis (second paragraph of Section 2).  Since the letters in a dot-matrix font do not form connected components, I am wondering if Tesseract's connected component analysis may be one reason that Tesseract struggles on the image attached.  

Is there a command to see how Tesseract performs connected component analysis on this image?

**Comment by Shreeshrii** [comment]
> I read in An Overview of the Tesseract OCR Engine that the first step in Tesseract's processing pipeline is a connected component analysis (second paragraph of Section 2).

That applies to base/legacy tesseract (3.0x). Tesseract4 is neural net/LSTM based though it still supports the legacy models.

I tried some finetuning using dot matrix fonts. While the image you gave is not recognized correctly, if split in separate lines the results are better, but only with --psm 8 (single word).

```
/home/ubuntu/TEST/ex_20_1 dotslayer
LOT#S18FO70

/home/ubuntu/TEST/ex_20_2 dotslayer
EXP:09/2020

```
![ex_20_2](https://user-images.githubusercontent.com/5095331/54854874-5d38c200-4d1a-11e9-9b86-4106dfd08b51.png)
![ex_20_1](https://user-images.githubusercontent.com/5095331/54854875-5dd15880-4d1a-11e9-96a1-b3212a62e983.png)

The traineddata file is available at https://github.com/Shreeshrii/tessdata_shreetest/blob/master/dotslayer.traineddata.

Note: I had also replied in forum with the same msg and files, maybe it didn't go thru because of attachments.
